flair/models/sequence_tagger_model.py

```diff
@@ -396,310 +303,217 @@ def forward(self, sentences: List[Sentence]):
-            # if self.use_word_dropout > 0.0:
-            #     sentence_tensor = self.word_dropout(sentence_tensor)
-            if self.use_locked_dropout > 0.0:
-                sentence_tensor = self.locked_dropout(sentence_tensor)
+        return scores
 
-        features = self.linear(sentence_tensor)
+    def _get_gold_labels(self, sentences: Union[List[DataPoint], DataPoint]):
+        """
+        Extracts gold labels from each sentence.
+        :param sentences: List of sentences in batch
+        """
+        tokens_per_sentence = [[token for token in sentence] for sentence in sentences]
+        labels = [[[token.get_tag(self.label_type).value] for token in sentence] for sentence in tokens_per_sentence]
+        labels = [token for sentence in labels for token in sentence]
 
-        return features
+        return labels
 
-    def _score_sentence(self, feats, tags, lens_):
+    def predict(
+        self,
+        sentences: Union[List[Sentence], Sentence],
+        mini_batch_size: int = 32,
+        return_probabilities_for_all_classes: bool = False,
+        verbose: bool = False,
+        label_name: Optional[str] = None,
+        return_loss=False,
+        embedding_storage_mode="none",
+    ):
+        """
+        Predicts labels for current batch with CRF or Softmax.
+        :param sentences: List of sentences in batch
+        :param mini_batch_size: batch size for test data
+        :param return_probabilities_for_all_classes: Whether to return probabilites for all classes
+        :param verbose: whether to use progress bar
+        :param label_name: which label to predict
+        :param return_loss: whether to return loss value
+        :param embedding_storage_mode: determines where to store embeddings - can be "gpu", "cpu" or None.
+        """
+        if label_name is None:
+            label_name = self.tag_type
 
-        start = torch.tensor([self.tag_dictionary.get_idx_for_item(START_TAG)], device=flair.device)
-        start = start[None, :].repeat(tags.shape[0], 1)
+        with torch.no_grad():
+            if not sentences:
+                return sentences
 
-        stop = torch.tensor([self.tag_dictionary.get_idx_for_item(STOP_TAG)], device=flair.device)
-        stop = stop[None, :].repeat(tags.shape[0], 1)
+            # make sure its a list
+            if not isinstance(sentences, list):
+                sentences = [sentences]
 
-        pad_start_tags = torch.cat([start, tags], 1)
-        pad_stop_tags = torch.cat([tags, stop], 1)
+            # filter empty sentences
+            if isinstance(sentences[0], Sentence):
+                sentences = [sentence for sentence in sentences if len(sentence) > 0]
 
-        for i in range(len(lens_)):
-            pad_stop_tags[i, lens_[i] :] = self.tag_dictionary.get_idx_for_item(STOP_TAG)
+            # order by length
+            reordered_sentences: List[Union[Sentence, str]] = sorted(sentences, key=lambda s: len(s), reverse=True)
```

Comment:
The Union doesn't make sense, as sentences is of type `List[Sentence]` we will always have `reordered_sentences: List[Sentence]` also, I think mypy is able to auto-infer the type of `reordered_sentences` so the typing might not be necessary

Reply:
also a good point, this is I think a leftover from times when str could also be passed to the predict function!